core/payment/src/processor.rs

```diff
@@ -123,41 +197,29 @@ impl DriverRegistry {
         address: &str,
         mode: AccountMode,
     ) -> Result<String, AccountNotRegistered> {
-        if let Some((driver, reg_mode)) = self
+        if let Some(details) = self
             .accounts
             .get(&(platform.to_owned(), address.to_owned()))
         {
-            if reg_mode.contains(mode) {
-                return Ok(driver.to_owned());
+            if details.mode.contains(mode) {
+                return Ok(details.driver.clone());
             }
         }
 
         // If it's recv-only mode or no-mode (i.e. checking status) we can use any driver that
         // supports the given platform and doesn't require init for receiving.
         if !mode.contains(AccountMode::SEND) {
```

Comment:
bit odd to check `if not send` to handle `if receive`.. will work till we get more modes :)

Reply:
The mode can be:
* Send
* Receive
* Send+Receive
* None

This conditional is specifically designed to capture Receive and None as stated by the comment above. It is assumed that if receiving funds doesn't require initialization then checking status will nit require it either.

Reply:
You are right, thanks for the explanation!